examples/basicauth/main.go

```diff
@@ -34,14 +35,14 @@ func main() {
 	}
 
 	client := github.NewClient(tp.Client())
-	user, _, err := client.Users.Get("")
+	user, _, err := client.Users.Get(context.Background(), "")
 
 	// Is this a two-factor auth error? If so, prompt for OTP and try again.
 	if _, ok := err.(*github.TwoFactorAuthError); err != nil && ok {
 		fmt.Print("\nGitHub OTP: ")
 		otp, _ := r.ReadString('\n')
 		tp.OTP = strings.TrimSpace(otp)
-		user, _, err = client.Users.Get("")
+		user, _, err = client.Users.Get(context.Background(), "")
```

Comment:
Hmmm... I'm not sure I agree with that.

Most clients that are not being used within a server will simply want `ctx := context.Background()` and as such is a fine example, especially for newbie Go devs. More experienced Go programmers will realize that they can replace the context with any one that they already have in operation and will have no problem swapping it out for their own.

But having said that, I don't feel really strongly about `TODO` vs `Background`.

I feel more strongly, though, about not duplicating the call and instead assigning to a `ctx` variable, as that is more idiomatic Go.